modules/go/main.go

```diff
@@ -81,10 +81,12 @@ func New(
 				// The specific version is dictated by Dagger's own requirement
 				// FIXME: make this optional with overlay support
 				"protoc~3.21.12",
+				"curl",
 			}}).
 			WithEnvVariable("GOLANG_VERSION", version).
 			WithEnvVariable("GOPATH", "/go").
 			WithEnvVariable("PATH", "${GOPATH}/bin:${PATH}", dagger.ContainerWithEnvVariableOpts{Expand: true}).
+			WithExec([]string{"sh", "-c", "go install gotest.tools/gotestsum@v1.12.0"}).
```

Comment:
nit: 

```suggestion
			WithExec([]string{"go", "install", "gotest.tools/gotestsum@v1.12.0"}).
```

We can also move this below the `WithMountedCache` calls, so that this can be part of populating the cache if there are shared dependencies.